### ISSUE TITLE
StringComparer.OrdinalIgnoreCase not ignoring case

### ISSUE THREAD

**Issue by fubar-coder** [dotnet/corefx#36881]
I got a strange problem and I guess that something changed between 2.1.505 (works) and 2.1.603 (fails).

```csharp
new HashSet<string>(new[] { "A" }, StringComparer.OrdinalIgnoreCase).Contains("a")
```

The expression above seems to return **false** on Travis CI when I use the .NET Core SDK 2.1.603 and **true** when I user .NET Core SDK 2.1.505.

The code gets executed in a context where the locale is set to `tr-TR`. I assume (just a wild guess here) that .NET Core cannot find information about the `tr-TR` locale and therefore compares case-sensitive.

Using `ToUpperInvariant()` doesn't seem to work either.

**Comment by danmosemsft** [comment]
@fubar-coder what OS configuration are you using? When you compare 2.1.603 and 2.1.505 are those on the exact same machine? If not can you please try that. 

@tarekgh

**Comment by tarekgh** [comment]
@fubar-coder I assume you are running on Linux. If so, this looks you machine is using C/POSIX locale as a default locale. These locales are not supporting casing. You can confirm that by dumping the values of the environment variables: 

```
LANG
LC_ALL
```

To fix that, you'll need to see the default to any other locale (e.g. en_US)

**Comment by fubar-coder** [comment]
@danmosemsft  It's a standard Travis CI build environment (Ubuntu 14.04), so I cannot ensure that it's the same **physical** machine, but it should always be the same virtual machine, because `dotnet-sdk-2.1` gets always installed by an additional script.

@tarekgh The problem doesn't seem to be the `en_US.UTF-8` locale, it instead occurs when I switch the current culture (for this single test) to `tr-TR`.

# lsb_release -a

```text
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty
```

# Environment variables

```bash
LC_ALL=en_US.UTF-8
LANG=en_US.UTF-8
```

They are indeed the same as described on the [Environment Variables](https://docs.travis-ci.com/user/environment-variables) page.

# mono --version

```text
Mono JIT compiler version 5.20.1.19 (tarball Thu Apr 11 09:05:42 UTC 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(600)
	Suspend:       hybrid
	GC:            sgen (concurrent by default)
```

# dotnet --info (2.1.505)

```text
.NET Core SDK (reflecting any global.json):
 Version:   2.1.505
 Commit:    b220231234

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  14.04
 OS Platform: Linux
 RID:         ubuntu.14.04-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.505/

Host (useful for support):
  Version: 2.1.10
  Commit:  9013a2af4b

.NET Core SDKs installed:
  2.1.505 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.10 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.10 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.10 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

# dotnet --info (2.1.603)

```text
.NET Core SDK (reflecting any global.json):
 Version:   2.1.603
 Commit:    ae71c68742

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  14.04
 OS Platform: Linux
 RID:         ubuntu.14.04-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.603/

Host (useful for support):
  Version: 2.1.10
  Commit:  9013a2af4b

.NET Core SDKs installed:
  2.1.603 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.10 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.10 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.10 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

# msbuild /version

```text
Microsoft (R) Build Engine version 16.0.0.0 for Mono
Copyright (C) Microsoft Corporation. All rights reserved.

16.0.0.0
```

**Comment by tarekgh** [comment]
@fubar-coder I'll try to setup  Ubuntu 14.04 LTS and will try to reproduce. 

Could you please isolate the issue by just creating a simple console app using the same version of the SDK and look if you see the same behavior?

**Comment by tarekgh** [comment]
I have tried the exact same setting:

````
tarek@tarek-Virtual-Machine:~/DotnetCoreApp$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty

tarek@tarek-Virtual-Machine:~/DotnetCoreApp$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.1.603
 Commit:    ae71c68742

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  14.04
 OS Platform: Linux
 RID:         ubuntu.14.04-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.603/

Host (useful for support):
  Version: 2.1.10
  Commit:  9013a2af4b

.NET Core SDKs installed:
  2.1.603 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.10 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.10 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.10 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

Then I have the code:

```C#
            CultureInfo.CurrentCulture = new CultureInfo("tr-TR");
            bool b = new HashSet<string>(new [] { "A" }, StringComparer.OrdinalIgnoreCase).Contains("a");
            Console.WriteLine($"{b}");
```

And then running the code: 

```
tarek@tarek-Virtual-Machine:~/DotnetCoreApp$ dotnet run
True
```
So I am not seeing any problem. It would be good if you try to isolate the issue by just running simple reproduce as I did with only net core console app. 

Also, in the script https://github.com/fluentmigrator/fluentmigrator/blob/master/.travis.yml I am seeing the following:

```
- dotnet publish src/FluentMigrator.Console/FluentMigrator.Console.csproj -c Release -r win7-x86 -o "$TRAVIS_BUILD_DIR/publish/tools/net461/x86" -p:Platform=x86 -p:TargetFramework=net461
- dotnet publish src/FluentMigrator.Console/FluentMigrator.Console.csproj -c Release -r win7-x64 -o "$TRAVIS_BUILD_DIR/publish/tools/net461/x64" -p:Platform=x64 -p:TargetFramework=net461
- dotnet publish src/FluentMigrator.Console/FluentMigrator.Console.csproj -c Release -r any -o "$TRAVIS_BUILD_DIR/publish/tools/net461/any" -p:Platform=AnyCPU -p:TargetFramework=net461
```

Why you are publishing for net461? just trying to understand how your project work.

**Comment by tarekgh** [comment]
One more note, when using StringComparer.OrdinalIgnoreCase, it is culture independent. so it doesn't really matter what culture you have.

**Comment by fubar-coder** [comment]
I don't know what it was, but the problem doesn't exist anymore when I use Ubuntu 16.04 on Travis CI. Maybe they fixed something or it's a problem with their 14.04 image.

Closing for now.
